rs/log_consumer/src/consumer.rs

```diff
@@ -0,0 +1,178 @@
+use std::time::Duration;
+
+use anyhow::Result;
+use log::{error, info};
+use rdkafka::config::ClientConfig;
+use rdkafka::consumer::{BaseConsumer, Consumer};
+use rdkafka::{Message, Offset, TopicPartitionList};
+use rmp_serde::decode;
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct LogMessage<T> {
+    pub payload: T,
+    pub topic: String,
+}
+
+pub struct LogConsumer {
+    inner: BaseConsumer,
+}
+
+impl LogConsumer {
+    pub fn new(broker: &str) -> Result<Self> {
+        let consumer: BaseConsumer = ClientConfig::new()
+            .set("bootstrap.servers", broker)
+            .set("group.id", "log_consumer")
+            // start from the beginning of the topic if no offset is stored
+            .set("auto.offset.reset", "earliest")
+            // manually commit offsets
+            .set("enable.auto.commit", "false")
+            .create()?;
+
+        Ok(LogConsumer { inner: consumer })
+    }
+
+    pub async fn consume_logs(&self) -> Result<usize> {
+        let mut processed_ops = 0;
+
+        while let Some(Ok(result)) = self.inner.poll(Duration::from_millis(500)) {
+            processed_ops += 1;
+            match decode::from_slice::<LogMessage<Vec<u8>>>(result.payload().unwrap()) {
+                Ok(wal_entry_message) => {
+                    let decoded: Value = decode::from_slice(&wal_entry_message.payload)?;
+                    let message = format!(
+                        "WAL offset: {}, partition: {}, payload: {:?}",
+                        result.offset(),
+                        result.partition(),
+                        decoded
+                    );
+                    info!("{}", message);
```

Comment:
put something like
```
#[cfg(debug_assertions)]
{
   let message = ... 
}
```
otherwise it will construct a log line every messaage, very expensive